### PR TITLE
[payment_paypal] end up on /shop/confirmation page at the end of the payment

### PR DESCRIPTION
**Pull request by rop-odoo**
Task: https://www.odoo.com/web?#id=35902&view_type=form&model=project.task&action=327&menu_id=4720
Pad: https://pad.odoo.com/p/r.c37d807abc28784a7995798e24d94f1e

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_paypal/controllers/main.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def _get_return_url(self, **post):
         return_url = post.pop('return_url', '')
         if not return_url:
             custom = json.loads(urls.url_unquote_plus(post.pop('custom', False) or post.pop('cm', False) or '{}'))
-            return_url = custom.get('return_url', '/')
+            return_url = custom.get('return_url', '/shop/confirmation')
```
Comment (tde-banana-odoo) [inline addons/payment_paypal/controllers/main.py]: Are you sure this is modular :rabbit2: ?

File: addons/payment_paypal/controllers/main.py [hunk 2]
```diff
@@ -112,5 +113,4 @@ def paypal_dpn(self, **post):
     def paypal_cancel(self, **post):
         """ When the user cancels its Paypal payment: GET on this route """
         _logger.info('Beginning Paypal cancel with post data %s', pprint.pformat(post))  # debug
-        return_url = self._get_return_url(**post)
-        return werkzeug.utils.redirect(return_url)
+        return werkzeug.utils.redirect('/shop/payment')
```
Comment (tde-banana-odoo) [inline addons/payment_paypal/controllers/main.py]: Are you sure this is modular :rabbit2: ?

File: addons/payment_paypal/models/payment.py [hunk 3]
```diff
@@ -38,6 +43,27 @@ class AcquirerPaypal(models.Model):
     fees_int_fixed = fields.Float(default=0.35)
     fees_int_var = fields.Float(default=3.9)
 
+    def _format_notify_message_data(self):
+        return _('''<div><i>Invalid/ In Progrogess/ Fail,</i>
```
Comment (tde-banana-odoo) [inline addons/payment_paypal/models/payment.py]: Progrogess ?

File: addons/payment/models/payment_acquirer.py [hunk 4]
```diff
@@ -115,6 +115,9 @@ class PaymentAcquirer(models.Model):
         'Error Message', translate=True,
         default='<i>Error,</i> Please be aware that an error occurred during the transaction. The order has been confirmed but will not be paid. Do not hesitate to contact us if you have any questions on the status of your order.',
         help='Message displayed, if error is occur during the payment process.')
+    notify_msg = fields.Html(
```
Comment (tde-banana-odoo) [inline addons/payment/models/payment_acquirer.py]: Shouldn't we use the pending_msg ? I don't see the difference.

File: addons/payment_paypal/controllers/main.py [hunk 5]
```diff
@@ -25,7 +25,10 @@ def _get_return_url(self, **post):
         return_url = post.pop('return_url', '')
         if not return_url:
             custom = json.loads(urls.url_unquote_plus(post.pop('custom', False) or post.pop('cm', False) or '{}'))
-            return_url = custom.get('return_url', '/')
+            if post.get('is_cancel'):
```
[the remark this answers is not in the document]
Comment (rop-odoo) [reply addons/payment_paypal/controllers/main.py]: Hello @tde-banana-odoo , user doing payment using paypal and user cancel the payment at that time this 72e41b0#diff-4f43da646770da97fdf6437e31f7b038R115 controller should be called and no any data get into post for that i need to set 72e41b0#diff-4f43da646770da97fdf6437e31f7b038R29 for cancel and for return url 72e41b0#diff-4f43da646770da97fdf6437e31f7b038R31 .

File: addons/payment_paypal/models/payment.py [hunk 6]
```diff
@@ -38,6 +43,26 @@ class AcquirerPaypal(models.Model):
     fees_int_fixed = fields.Float(default=0.35)
     fees_int_var = fields.Float(default=3.9)
 
+    def _format_notify_message_data(self):
```
[the remark this answers is not in the document]
Comment (rop-odoo) [reply addons/payment_paypal/models/payment.py]: Hello @tde-banana-odoo , As per discussion with @fgi-odoo , When IPN mode is activated and payment process done successfully but notification not comes from paypal at that time user should redirect on confirmation page. for that need to change the pending message for indicate end user to payment processed successfully.

**Comment by hmo-odoo** [comment]
cancelled

**Pull request closed without merging**